src/ts/editorcanvas/CanvasVM.ts

```diff
@@ -232,26 +251,62 @@ module editorcanvas {
             this.stage.y = Math.max(newDimensions.y / 2, this.canvasDiv.clientHeight / 2) - this.canvasDiv.scrollTop;
         }
 
-        public redrawCanvas() {
-
-            var toDraw : Array<createjs.DisplayObject> = [];
+        private areAllAssetsLoaded() : boolean {
+            return this.numAssetsToLoad === this.numAssetsLoaded;
+        }
 
-            this.data.forEach((entity) => {
-                toDraw.push(this.entityDrawerService.getEntityDisplayable(entity));
-            });
+        /**
+         * Called every time the canvas needs to be updated and redrawn.
+         */
+        redrawCanvas() {
+            if (!this.areAllAssetsLoaded()) {
+                return;
+            }
 
+            this.disableInterpolation();
             this.clear();
+            this.addDrawnElementsToStage();
+            this.stage.update();
+        }
+
+        /**
+         * Disables interpolation so scaling is done via the nearest neighbor approach.
+         */
+        private disableInterpolation() {
+            var context : any = (<HTMLCanvasElement>this._stage.canvas).getContext("2d");
+            context.imageSmoothingEnabled = false;
+        }
+
+        /**
+         * Collect the entity's DrawableComponent's drawables into a flat array
+         * that is sorted by the render priorities of the drawables.
+         */
+        private collectEntityDrawables() : Array<createjs.DisplayObject> {
+            var toDraw : Array<createjs.DisplayObject> = [];
+            var entitiesByPriority = this._context.getSharedObject(services.EntityRenderSortService).getEntitiesByPriority();
+            entitiesByPriority.forEach((entityKey : string) => {
```

Comment:
An entity will only appear in the entitiesByPriority array if it has a drawable component, so it will only render in that case. However, other components can have a rendered portion (such as collision component)